installer/client/src/core/util/composable/useProgress.ts

```diff
@@ -37,7 +37,10 @@ export default function useProgress() {
   const increment = () => {
     setTimeout(() => {
       if (progress.value === 100 || !working) return;
-      progress.value = generator.next().value * 100;
+      // to prevent progress.value reaching 100 prematurely
+      if (progress.value < 99) {
+        progress.value = generator.next().value * 100;
+      }
```

Comment:
when migrations are slow, `progress.value` will eventually reach 100 since it is always being incremented. added a condition to prevent it